src/prefect/agent/agent.py

```diff
@@ -36,7 +37,11 @@ class Agent:
     """
 
     def __init__(self) -> None:
-        self.client = Client(api_token=config.cloud.agent.get("auth_token"))
+
+        token = config.cloud.agent.get("auth_token")
+
+        self.client = Client(api_token=token)
+        self._verify_token(token)
```

Comment:
question: should we verify the provided `token` _or_ the token the client intends to use?
```python
self._verify_token(self.client._api_token)
```
? whichever is more robust and provides the most informative error, just wanted to ask in case there is an edge case lurking in here

Reply:
If we go off of `client._api_token` then we won't catch the case in where an agent token is empty because it'll default to the `config.cloud.api_token`

Reply:
Since this is an issue of if not provided use a USER now and None in the deployed flow

Reply:
We'll still catch that it's not a `RUNNER` but it would be nice to still catch if nothing is provided

Reply:
Got it, ok cool